johnzon-mapper/src/main/java/org/apache/johnzon/mapper/Mappings.java

```diff
@@ -332,39 +329,50 @@ public static boolean isPrimitive(final Type type) {
         return false;
     }
 
-    public ClassMapping getClassMapping(final Type clazz) {
+    public TypeMapping getClassMapping(final Type clazz) {
         return classes.get(clazz);
     }
 
-    public ClassMapping findOrCreateClassMapping(final Type clazz) {
-        ClassMapping classMapping = classes.get(clazz);
+    public TypeMapping findOrCreateTypeMapping(final Type clazz) {
+        return findOrCreateTypeMapping(clazz, false);
+    }
+
+    public TypeMapping findOrCreateTypeMapping(final Type clazz, final boolean applyObjectConverter) {
+        TypeMapping classMapping = classes.get(clazz);
         if (classMapping == null) {
-            if (!Class.class.isInstance(clazz) || Map.class.isAssignableFrom(Class.class.cast(clazz))) {
-                return null;
+            if (Class.class.isInstance(clazz)) {
+                if (!Map.class.isAssignableFrom(Class.class.cast(clazz))){
+                    classMapping = createTypeMapping(Class.class.cast(clazz), applyObjectConverter);
+                }
+            } else if (ParameterizedType.class.isInstance(clazz)){
+                classMapping = findOrCreateTypeMapping(ParameterizedType.class.cast(clazz).getRawType(), applyObjectConverter);
+            } else if (TypeVariable.class.isInstance(clazz)){
+                TypeVariable cast = TypeVariable.class.cast(clazz);
+                if (Objects.nonNull(cast.getAnnotatedBounds()) && cast.getAnnotatedBounds().length > 0){
+                    classMapping = findOrCreateTypeMapping(TypeVariable.class.cast(clazz).getAnnotatedBounds()[0].getType(), applyObjectConverter);
```

Comment:
probably store it in the TypeVariableMapping too?

Reply:
I removed multiple type of mapping after What we said. Do we need them?